FFMediaToolkit/Decoding/Internal/InputContainer.cs

```diff
@@ -156,30 +149,33 @@ private static InputContainer MakeContainer(string url, MediaOptions options, AV
         /// <param name="options">The <see cref="MediaOptions"/> object.</param>
         private void OpenStreams(MediaOptions options)
         {
-            if (options.StreamsToLoad == MediaMode.AudioVideo || options.StreamsToLoad == MediaMode.Video)
-                Video = DecoderFactory.OpenVideo(this, options);
-
-            if (options.StreamsToLoad == MediaMode.AudioVideo || options.StreamsToLoad == MediaMode.Audio)
-                Audio = DecoderFactory.OpenAudio(this, options);
-
-            // Requests for the first packet.
-            if (Video != null && Audio != null)
-                ReadPacket();
-            else if (Video != null)
-                GetPacketFromStream(Video.Info.Index);
-            else if (Audio != null)
-                GetPacketFromStream(Audio.Info.Index);
-            else
-                return;
-            canReusePacket = true;
+            for (int i = 0; i < Pointer->nb_streams; i++)
+            {
+                var stream = Pointer->streams[i];
+                var mode = (MediaMode)(1 << (int)stream->codec->codec_type);
```

Comment:
So essentially to convert the AVCodecType into the MediaMode type, I am doing a couple of things here.  Essentially, to make the MediaMode enum less redundant, I converted it into a "Flags" enumeration, where the first bit represents a value that includes video streams, and the second bit represents a value that includes audio streams.  I set these to 1 and 2 respectively (01 and 10 in binary).  I did so in this order very purposefully because then I can use a bit shift to convert the ordinal values of AVCodecType into the positional bit flag values for MediaMode.  
In retrospect this is a fairly convoluted solution, but its ultimately better than a clunky conversion with a bunch of branching statements.

Reply:
Well, it's using the values of that externally defined enum to construct values of different enum in a rather non-transparent way, I think it would be ok if:
- the code were put somewhere in same file as definition of MediaMode
- MediaMode enum values were themselves defined by manipulating original AvCodeType constants, so its definition is explicitly corresponding it to and it's clear how the conversions can be done

Reply:
Yes, I thought about this a little bit as well, especially the second point.  Instead of defining MediaMode constants in terms of "magic numbers" I could easily use a bitwise shift operator directly in the definition to make the association between the two enumerations more clear, and write an extension class that handles the conversion.